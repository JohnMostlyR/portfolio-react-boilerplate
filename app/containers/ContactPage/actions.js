/*
 *
 * ContactPage actions
 *
 */

import { SEND_FORM, SEND_FORM_SUCCESS, SEND_FORM_ERROR } from './constants';

export function sendForm(field) {
  return {
    type: SEND_FORM,
    status: 'sending',
    field,
  };
}

export function sendFormSuccess() {
  return {
    type: SEND_FORM_SUCCESS,
    status: 'success',
  };
}

export function sendFormError(error) {
  return {
    type: SEND_FORM_ERROR,
    status: 'error',
    error,
  };
}
