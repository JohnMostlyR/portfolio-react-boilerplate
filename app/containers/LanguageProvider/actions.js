/*
 *
 * LanguageProvider actions
 *
 */

import { CHANGE_LANGUAGE } from './constants';

export function changeLanguage(languageLocale) {
  return {
    type: CHANGE_LANGUAGE,
    locale: languageLocale,
  };
}
