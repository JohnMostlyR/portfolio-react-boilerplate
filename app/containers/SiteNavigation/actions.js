/*
 *
 * SiteNavigation actions
 *
 */

import {
  SET_SITE_NAVIGATION_IS_AT_SCREEN_TOP,
  SET_SITE_NAVIGATION_OFFSET_HEIGHT,
  SET_SITE_NAVIGATION_TOP_POSITION,
} from './constants';

export function setSiteNavigationTopPosition(topPosition) {
  return {
    type: SET_SITE_NAVIGATION_TOP_POSITION,
    topPosition,
  };
}

export function setSiteNavigationOffsetHeight(offsetHeight) {
  return {
    type: SET_SITE_NAVIGATION_OFFSET_HEIGHT,
    offsetHeight,
  };
}

export function setSiteNavigationIsAtScreenTop(isAtScreenTop) {
  return {
    type: SET_SITE_NAVIGATION_IS_AT_SCREEN_TOP,
    isAtScreenTop,
  };
}
