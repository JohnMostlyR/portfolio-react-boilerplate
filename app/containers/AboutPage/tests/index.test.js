import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import createHistory from 'history/createMemoryHistory';
import marked from 'marked';
import Parser from 'html-react-parser';

import { AboutPage, mapDispatchToProps } from '../index';
import { loadContent } from '../actions';
import configureStore from '../../../configureStore';
import PageContent from '../../../components/PageContent';

// enable fake timers
jest.useFakeTimers();

// set options for marked
marked.options({
  breaks: true,
});

describe('<AboutPage />', () => {
  it('should render and match the snapshot', () => {
    const getContent = jest.fn();
    const wrapper = shallow(<AboutPage getContent={getContent} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('fetch content from external source', () => {
    const componentWillReceivePropsSpy = jest.spyOn(
      AboutPage.prototype,
      'componentWillReceiveProps',
    );
    const handleIsLoadingSpy = jest.spyOn(
      AboutPage.prototype,
      'handleIsLoading',
    );
    const setIsLoadingSpy = jest.spyOn(AboutPage.prototype, 'setIsLoading');
    const getContent = jest.fn();
    let isLoadingProp = false;
    let wrapper = {};

    afterEach(() => {
      wrapper = null;
      jest.clearAllTimers();
      jest.clearAllMocks();
    });

    it('should set the "isLoading" state when the "isLoading" prop is true for more than 200ms', () => {
      wrapper = shallow(
        <AboutPage getContent={getContent} isLoading={false} />,
      );
      expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(0);
      expect(handleIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);

      isLoadingProp = true;

      wrapper.setProps({ isLoading: isLoadingProp });

      expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(1);
      expect(handleIsLoadingSpy).toHaveBeenCalledTimes(1);
      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().isLoading).toBe(false);

      jest.runTimersToTime(300);

      expect(setIsLoadingSpy).toHaveBeenCalledTimes(1);
      expect(wrapper.state().isLoading).toBe(true);
    });

    it('should not set the "isLoading" state when the "isLoading" prop is true for less than 200ms', () => {
      wrapper = shallow(
        <AboutPage getContent={getContent} isLoading={false} />,
      );
      expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(0);
      expect(handleIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);

      isLoadingProp = true;

      wrapper.setProps({ isLoading: isLoadingProp });

      expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(1);
      expect(handleIsLoadingSpy).toHaveBeenCalledTimes(1);
      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().isLoading).toBe(false);

      jest.runTimersToTime(100);

      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().isLoading).toBe(false);
    });
  });

  describe('fetching content from external source', () => {
    const history = createHistory();
    const store = configureStore({}, history);
    const getContent = jest.fn();
    let wrapper = {};

    afterEach(() => {
      wrapper = null;
      jest.clearAllMocks();
    });

    it('should call getContent() to load external data', () => {
      mount(
        <Provider store={store}>
          <IntlProvider locale="en">
            <AboutPage getContent={getContent} />
          </IntlProvider>
        </Provider>,
      );

      expect(getContent).toBeCalled();
    });

    it('should show an error message when content could not be loaded', () => {
      wrapper = mount(
        <Provider store={store}>
          <IntlProvider locale="en">
            <AboutPage getContent={getContent} error />
          </IntlProvider>
        </Provider>,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should handle case where invalid content is received without any error', () => {
      wrapper = mount(
        <Provider store={store}>
          <IntlProvider locale="en">
            <AboutPage
              getContent={getContent}
              isLoading={false}
              error={false}
              skillsText={[]}
            />
          </IntlProvider>
        </Provider>,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should show the fetched content', () => {
      const fixture = String.raw`Test Content\nNew Line`;
      wrapper = mount(
        <Provider store={store}>
          <IntlProvider locale="en">
            <AboutPage
              getContent={getContent}
              isLoading={false}
              error={false}
              aboutMeText={Parser(marked(fixture))}
            />
          </IntlProvider>
        </Provider>,
      );

      expect(wrapper.contains(PageContent)).toEqual(true);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
    describe('getContent', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const props = mapDispatchToProps(dispatch);
        expect(props.getContent).toBeDefined();
      });

      it('should dispatch loadContent when called', () => {
        const dispatch = jest.fn();
        const props = mapDispatchToProps(dispatch);
        props.getContent();
        expect(dispatch).toHaveBeenCalledWith(loadContent());
      });
    });
  });
});
