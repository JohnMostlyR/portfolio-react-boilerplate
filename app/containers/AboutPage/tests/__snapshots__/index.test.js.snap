// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage /> fetching content from external source should handle case where invalid content is received without any error 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <IntlProvider
    locale="en"
  >
    <AboutPage
      error={false}
      getContent={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      isLoading={false}
      skillsText={Array []}
    >
      <InjectIntl(HeadGear)
        messages={
          Object {
            "metaDescription": Object {
              "defaultMessage": "A short description about me and my professional life",
              "id": "portfolio.page.AboutPage.meta.description",
            },
            "pageTitle": Object {
              "defaultMessage": "ABOUT ME",
              "id": "portfolio.page.AboutPage.pageTitle",
            },
            "title": Object {
              "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
              "id": "portfolio.page.AboutPage.title",
            },
          }
        }
        path="/about"
      >
        <HeadGear
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {},
              "now": [Function],
              "textComponent": "span",
            }
          }
          messages={
            Object {
              "metaDescription": Object {
                "defaultMessage": "A short description about me and my professional life",
                "id": "portfolio.page.AboutPage.meta.description",
              },
              "pageTitle": Object {
                "defaultMessage": "ABOUT ME",
                "id": "portfolio.page.AboutPage.pageTitle",
              },
              "title": Object {
                "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
                "id": "portfolio.page.AboutPage.title",
              },
            }
          }
          path="/about"
        >
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
            link={
              Array [
                Object {
                  "href": "https://meester-johan.info/about/",
                  "rel": "canonical",
                },
                Object {
                  "href": "https://meester-johan.info/about/",
                  "hreflang": "x-default",
                  "rel": "alternate",
                },
              ]
            }
            meta={
              Array [
                Object {
                  "content": "A short description about me and my professional life",
                  "name": "description",
                },
                Object {
                  "content": "A short description about me and my professional life",
                  "property": "og:description",
                },
                Object {
                  "content": "Johan Meester : Portfolio : Front-End Developer : ABOUT ME",
                  "property": "og:title",
                },
                Object {
                  "content": "https://meester-johan.info/about/",
                  "property": "og.url",
                },
              ]
            }
            title="Johan Meester : Portfolio : Front-End Developer : ABOUT ME"
          >
            <SideEffect(NullComponent)
              defer={true}
              encodeSpecialCharacters={true}
              link={
                Array [
                  Object {
                    "href": "https://meester-johan.info/about/",
                    "rel": "canonical",
                  },
                  Object {
                    "href": "https://meester-johan.info/about/",
                    "hreflang": "x-default",
                    "rel": "alternate",
                  },
                ]
              }
              meta={
                Array [
                  Object {
                    "content": "A short description about me and my professional life",
                    "name": "description",
                  },
                  Object {
                    "content": "A short description about me and my professional life",
                    "property": "og:description",
                  },
                  Object {
                    "content": "Johan Meester : Portfolio : Front-End Developer : ABOUT ME",
                    "property": "og:title",
                  },
                  Object {
                    "content": "https://meester-johan.info/about/",
                    "property": "og.url",
                  },
                ]
              }
              title="Johan Meester : Portfolio : Front-End Developer : ABOUT ME"
            >
              <NullComponent
                defer={true}
                encodeSpecialCharacters={true}
                link={
                  Array [
                    Object {
                      "href": "https://meester-johan.info/about/",
                      "rel": "canonical",
                    },
                    Object {
                      "href": "https://meester-johan.info/about/",
                      "hreflang": "x-default",
                      "rel": "alternate",
                    },
                  ]
                }
                meta={
                  Array [
                    Object {
                      "content": "A short description about me and my professional life",
                      "name": "description",
                    },
                    Object {
                      "content": "A short description about me and my professional life",
                      "property": "og:description",
                    },
                    Object {
                      "content": "Johan Meester : Portfolio : Front-End Developer : ABOUT ME",
                      "property": "og:title",
                    },
                    Object {
                      "content": "https://meester-johan.info/about/",
                      "property": "og.url",
                    },
                  ]
                }
                title="Johan Meester : Portfolio : Front-End Developer : ABOUT ME"
              />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
        </HeadGear>
      </InjectIntl(HeadGear)>
    </AboutPage>
  </IntlProvider>
</Provider>
`;

exports[`<AboutPage /> fetching content from external source should show an error message when content could not be loaded 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <IntlProvider
    locale="en"
  >
    <AboutPage
      error={true}
      getContent={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      <InjectIntl(HeadGear)
        messages={
          Object {
            "metaDescription": Object {
              "defaultMessage": "A short description about me and my professional life",
              "id": "portfolio.page.AboutPage.meta.description",
            },
            "pageTitle": Object {
              "defaultMessage": "ABOUT ME",
              "id": "portfolio.page.AboutPage.pageTitle",
            },
            "title": Object {
              "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
              "id": "portfolio.page.AboutPage.title",
            },
          }
        }
        path="/about"
      >
        <HeadGear
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {},
              "now": [Function],
              "textComponent": "span",
            }
          }
          messages={
            Object {
              "metaDescription": Object {
                "defaultMessage": "A short description about me and my professional life",
                "id": "portfolio.page.AboutPage.meta.description",
              },
              "pageTitle": Object {
                "defaultMessage": "ABOUT ME",
                "id": "portfolio.page.AboutPage.pageTitle",
              },
              "title": Object {
                "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
                "id": "portfolio.page.AboutPage.title",
              },
            }
          }
          path="/about"
        >
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
            link={
              Array [
                Object {
                  "href": "https://meester-johan.info/about/",
                  "rel": "canonical",
                },
                Object {
                  "href": "https://meester-johan.info/about/",
                  "hreflang": "x-default",
                  "rel": "alternate",
                },
              ]
            }
            meta={
              Array [
                Object {
                  "content": "A short description about me and my professional life",
                  "name": "description",
                },
                Object {
                  "content": "A short description about me and my professional life",
                  "property": "og:description",
                },
                Object {
                  "content": "Johan Meester : Portfolio : Front-End Developer : ABOUT ME",
                  "property": "og:title",
                },
                Object {
                  "content": "https://meester-johan.info/about/",
                  "property": "og.url",
                },
              ]
            }
            title="Johan Meester : Portfolio : Front-End Developer : ABOUT ME"
          >
            <SideEffect(NullComponent)
              defer={true}
              encodeSpecialCharacters={true}
              link={
                Array [
                  Object {
                    "href": "https://meester-johan.info/about/",
                    "rel": "canonical",
                  },
                  Object {
                    "href": "https://meester-johan.info/about/",
                    "hreflang": "x-default",
                    "rel": "alternate",
                  },
                ]
              }
              meta={
                Array [
                  Object {
                    "content": "A short description about me and my professional life",
                    "name": "description",
                  },
                  Object {
                    "content": "A short description about me and my professional life",
                    "property": "og:description",
                  },
                  Object {
                    "content": "Johan Meester : Portfolio : Front-End Developer : ABOUT ME",
                    "property": "og:title",
                  },
                  Object {
                    "content": "https://meester-johan.info/about/",
                    "property": "og.url",
                  },
                ]
              }
              title="Johan Meester : Portfolio : Front-End Developer : ABOUT ME"
            >
              <NullComponent
                defer={true}
                encodeSpecialCharacters={true}
                link={
                  Array [
                    Object {
                      "href": "https://meester-johan.info/about/",
                      "rel": "canonical",
                    },
                    Object {
                      "href": "https://meester-johan.info/about/",
                      "hreflang": "x-default",
                      "rel": "alternate",
                    },
                  ]
                }
                meta={
                  Array [
                    Object {
                      "content": "A short description about me and my professional life",
                      "name": "description",
                    },
                    Object {
                      "content": "A short description about me and my professional life",
                      "property": "og:description",
                    },
                    Object {
                      "content": "Johan Meester : Portfolio : Front-End Developer : ABOUT ME",
                      "property": "og:title",
                    },
                    Object {
                      "content": "https://meester-johan.info/about/",
                      "property": "og.url",
                    },
                  ]
                }
                title="Johan Meester : Portfolio : Front-End Developer : ABOUT ME"
              />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
        </HeadGear>
      </InjectIntl(HeadGear)>
      <ContentLoadingIndicator
        show={true}
        showError={true}
      >
        <Article
          show={true}
        >
          <article
            className="Article-iozEAH ddRhqm"
          >
            <Header>
              <header>
                <SpeechBubble
                  arrowHeight="4vh"
                  arrowPosition="bottom-right"
                  backgroundColor="#465a65"
                  isGhost={false}
                  padding="1.5rem"
                  showArrowBreakpoint="750px"
                >
                  <Container
                    showArrowBreakpoint="750px"
                  >
                    <div
                      className="Container-gZqETv jUyUSn"
                    >
                      <Bubble
                        arrowPosition="bottom-right"
                        backgroundColor="#465a65"
                        isGhost={false}
                        padding="1.5rem"
                        showArrowBreakpoint="750px"
                      >
                        <div
                          className="Bubble-cYQkeS ixgWcF"
                        >
                          <Content>
                            <div
                              className="Content-zwkXZ gghqRA"
                            >
                              <H2
                                isError={true}
                              >
                                <h2
                                  className="H2-eUbBFj hYXgfx"
                                >
                                  <FormattedMessage
                                    defaultMessage="Terribly sorry, but the content cannot be loaded at this time"
                                    id="portfolio.component.ContentLoadingIndicator.error"
                                    values={Object {}}
                                  >
                                    <span>
                                      Terribly sorry, but the content cannot be loaded at this time
                                    </span>
                                  </FormattedMessage>
                                </h2>
                              </H2>
                              <SVGIcon
                                height="2em"
                                viewBox="0 0 512 512"
                                width="2em"
                              >
                                <svg
                                  aria-hidden="true"
                                  className="SVGIcon-ixpDyI cxenSi"
                                  focusable="false"
                                  height="2em"
                                  viewBox="0 0 512 512"
                                  width="2em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M304.083 405.907c4.686 4.686 4.686 12.284 0 16.971l-44.674 44.674c-59.263 59.262-155.693 59.266-214.961 0-59.264-59.265-59.264-155.696 0-214.96l44.675-44.675c4.686-4.686 12.284-4.686 16.971 0l39.598 39.598c4.686 4.686 4.686 12.284 0 16.971l-44.675 44.674c-28.072 28.073-28.072 73.75 0 101.823 28.072 28.072 73.75 28.073 101.824 0l44.674-44.674c4.686-4.686 12.284-4.686 16.971 0l39.597 39.598zm-56.568-260.216c4.686 4.686 12.284 4.686 16.971 0l44.674-44.674c28.072-28.075 73.75-28.073 101.824 0 28.072 28.073 28.072 73.75 0 101.823l-44.675 44.674c-4.686 4.686-4.686 12.284 0 16.971l39.598 39.598c4.686 4.686 12.284 4.686 16.971 0l44.675-44.675c59.265-59.265 59.265-155.695 0-214.96-59.266-59.264-155.695-59.264-214.961 0l-44.674 44.674c-4.686 4.686-4.686 12.284 0 16.971l39.597 39.598zm234.828 359.28l22.627-22.627c9.373-9.373 9.373-24.569 0-33.941L63.598 7.029c-9.373-9.373-24.569-9.373-33.941 0L7.029 29.657c-9.373 9.373-9.373 24.569 0 33.941l441.373 441.373c9.373 9.372 24.569 9.372 33.941 0z"
                                  />
                                </svg>
                              </SVGIcon>
                            </div>
                          </Content>
                        </div>
                      </Bubble>
                      <Arrow
                        arrowPosition="bottom-right"
                        backgroundColor="#465a65"
                        height="4vh"
                        isGhost={false}
                        showBreakpoint="750px"
                      >
                        <Arrow__SVG
                          arrowPosition="bottom-right"
                          height="4vh"
                          showBreakpoint="750px"
                        >
                          <svg
                            className="Arrow__SVG-iWxuxU jYKhgf"
                            focusable="false"
                            height="4vh"
                            preserveAspectRatio="xMinYMid meet"
                            viewBox="0 0 848 640"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M424.2 424.4L526.9 640V398.7"
                              fill="#5b7583"
                              stroke="none"
                              strokeWidth="24"
                            />
                            <path
                              d="M848 318.3L424.2 424.4V158.9"
                              fill="#465a65"
                              stroke="none"
                              strokeWidth="24"
                            />
                            <path
                              d="M0 0l848 318.3L636.4 0H0z"
                              fill="#5b7583"
                              stroke="none"
                              strokeWidth="24"
                            />
                          </svg>
                        </Arrow__SVG>
                      </Arrow>
                    </div>
                  </Container>
                </SpeechBubble>
              </header>
            </Header>
          </article>
        </Article>
      </ContentLoadingIndicator>
    </AboutPage>
  </IntlProvider>
</Provider>
`;

exports[`<AboutPage /> fetching content from external source should show the fetched content 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <IntlProvider
    locale="en"
  >
    <AboutPage
      aboutMeText={
        Array [
          <p>
            Test Content\\nNew Line
          </p>,
          "
",
        ]
      }
      error={false}
      getContent={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      isLoading={false}
    >
      <InjectIntl(HeadGear)
        messages={
          Object {
            "metaDescription": Object {
              "defaultMessage": "A short description about me and my professional life",
              "id": "portfolio.page.AboutPage.meta.description",
            },
            "pageTitle": Object {
              "defaultMessage": "ABOUT ME",
              "id": "portfolio.page.AboutPage.pageTitle",
            },
            "title": Object {
              "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
              "id": "portfolio.page.AboutPage.title",
            },
          }
        }
        path="/about"
      >
        <HeadGear
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {},
              "now": [Function],
              "textComponent": "span",
            }
          }
          messages={
            Object {
              "metaDescription": Object {
                "defaultMessage": "A short description about me and my professional life",
                "id": "portfolio.page.AboutPage.meta.description",
              },
              "pageTitle": Object {
                "defaultMessage": "ABOUT ME",
                "id": "portfolio.page.AboutPage.pageTitle",
              },
              "title": Object {
                "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
                "id": "portfolio.page.AboutPage.title",
              },
            }
          }
          path="/about"
        >
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
            link={
              Array [
                Object {
                  "href": "https://meester-johan.info/about/",
                  "rel": "canonical",
                },
                Object {
                  "href": "https://meester-johan.info/about/",
                  "hreflang": "x-default",
                  "rel": "alternate",
                },
              ]
            }
            meta={
              Array [
                Object {
                  "content": "A short description about me and my professional life",
                  "name": "description",
                },
                Object {
                  "content": "A short description about me and my professional life",
                  "property": "og:description",
                },
                Object {
                  "content": "Johan Meester : Portfolio : Front-End Developer : ABOUT ME",
                  "property": "og:title",
                },
                Object {
                  "content": "https://meester-johan.info/about/",
                  "property": "og.url",
                },
              ]
            }
            title="Johan Meester : Portfolio : Front-End Developer : ABOUT ME"
          >
            <SideEffect(NullComponent)
              defer={true}
              encodeSpecialCharacters={true}
              link={
                Array [
                  Object {
                    "href": "https://meester-johan.info/about/",
                    "rel": "canonical",
                  },
                  Object {
                    "href": "https://meester-johan.info/about/",
                    "hreflang": "x-default",
                    "rel": "alternate",
                  },
                ]
              }
              meta={
                Array [
                  Object {
                    "content": "A short description about me and my professional life",
                    "name": "description",
                  },
                  Object {
                    "content": "A short description about me and my professional life",
                    "property": "og:description",
                  },
                  Object {
                    "content": "Johan Meester : Portfolio : Front-End Developer : ABOUT ME",
                    "property": "og:title",
                  },
                  Object {
                    "content": "https://meester-johan.info/about/",
                    "property": "og.url",
                  },
                ]
              }
              title="Johan Meester : Portfolio : Front-End Developer : ABOUT ME"
            >
              <NullComponent
                defer={true}
                encodeSpecialCharacters={true}
                link={
                  Array [
                    Object {
                      "href": "https://meester-johan.info/about/",
                      "rel": "canonical",
                    },
                    Object {
                      "href": "https://meester-johan.info/about/",
                      "hreflang": "x-default",
                      "rel": "alternate",
                    },
                  ]
                }
                meta={
                  Array [
                    Object {
                      "content": "A short description about me and my professional life",
                      "name": "description",
                    },
                    Object {
                      "content": "A short description about me and my professional life",
                      "property": "og:description",
                    },
                    Object {
                      "content": "Johan Meester : Portfolio : Front-End Developer : ABOUT ME",
                      "property": "og:title",
                    },
                    Object {
                      "content": "https://meester-johan.info/about/",
                      "property": "og.url",
                    },
                  ]
                }
                title="Johan Meester : Portfolio : Front-End Developer : ABOUT ME"
              />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
        </HeadGear>
      </InjectIntl(HeadGear)>
      <PageContent
        content={
          Array [
            <p>
              Test Content\\nNew Line
            </p>,
            "
",
          ]
        }
        title={
          <FormattedMessage
            defaultMessage="ABOUT ME"
            id="portfolio.page.AboutPage.pageTitle"
            values={Object {}}
          />
        }
      >
        <Wrapper
          animate={true}
        >
          <div
            className="Wrapper-ThVjl ipiXwR"
          >
            <PageHeader
              arrowPosition="bottom-left"
            >
              <StyledHeader>
                <header
                  className="StyledHeader-fJIKuc cuJATB"
                >
                  <SpeechBubble
                    arrowHeight="4vh"
                    arrowPosition="bottom-left"
                    backgroundColor="#465a65"
                    isGhost={false}
                    makeAppear={true}
                    showArrowBreakpoint="750px"
                  >
                    <Container
                      makeAppear={true}
                      showArrowBreakpoint="750px"
                    >
                      <div
                        className="Container-gZqETv hayOh"
                      >
                        <Bubble
                          arrowPosition="bottom-left"
                          backgroundColor="#465a65"
                          isGhost={false}
                          showArrowBreakpoint="750px"
                        >
                          <div
                            className="Bubble-cYQkeS errvuT"
                          >
                            <H2
                              color="#aedd2c"
                            >
                              <h2
                                className="H2-eUbBFj iAWPFe"
                                color="#aedd2c"
                              >
                                <FormattedMessage
                                  defaultMessage="ABOUT ME"
                                  id="portfolio.page.AboutPage.pageTitle"
                                  values={Object {}}
                                >
                                  <span>
                                    ABOUT ME
                                  </span>
                                </FormattedMessage>
                              </h2>
                            </H2>
                          </div>
                        </Bubble>
                        <Arrow
                          arrowPosition="bottom-left"
                          backgroundColor="#465a65"
                          height="4vh"
                          isGhost={false}
                          showBreakpoint="750px"
                        >
                          <Arrow__SVG
                            arrowPosition="bottom-left"
                            height="4vh"
                            showBreakpoint="750px"
                          >
                            <svg
                              className="Arrow__SVG-iWxuxU iIEjSv"
                              focusable="false"
                              height="4vh"
                              preserveAspectRatio="xMinYMid meet"
                              viewBox="0 0 848 640"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M424.2 424.4L526.9 640V398.7"
                                fill="#5b7583"
                                stroke="none"
                                strokeWidth="24"
                              />
                              <path
                                d="M848 318.3L424.2 424.4V158.9"
                                fill="#465a65"
                                stroke="none"
                                strokeWidth="24"
                              />
                              <path
                                d="M0 0l848 318.3L636.4 0H0z"
                                fill="#5b7583"
                                stroke="none"
                                strokeWidth="24"
                              />
                            </svg>
                          </Arrow__SVG>
                        </Arrow>
                      </div>
                    </Container>
                  </SpeechBubble>
                </header>
              </StyledHeader>
            </PageHeader>
            <ContentWrapper>
              <div
                className="ContentWrapper-fNtIWq jXZlqv"
              >
                <SpeechBubble
                  arrowHeight="9vh"
                  arrowPosition="bottom-right"
                  backgroundColor="#aedd2c"
                  color="#000"
                  isGhost={false}
                  makeAppear={false}
                  maxWidth="30em"
                  padding="1.5rem"
                  showArrowBreakpoint="750px"
                >
                  <Container
                    makeAppear={false}
                    maxWidth="30em"
                    showArrowBreakpoint="750px"
                  >
                    <div
                      className="Container-gZqETv jsqDeB"
                    >
                      <Bubble
                        arrowPosition="bottom-right"
                        backgroundColor="#aedd2c"
                        color="#000"
                        isGhost={false}
                        padding="1.5rem"
                        showArrowBreakpoint="750px"
                      >
                        <div
                          className="Bubble-cYQkeS bOhOYj"
                          color="#000"
                        >
                          <p
                            key="0"
                          >
                            Test Content\\nNew Line
                          </p>
                          

                        </div>
                      </Bubble>
                      <Arrow
                        arrowPosition="bottom-right"
                        backgroundColor="#aedd2c"
                        height="9vh"
                        isGhost={false}
                        showBreakpoint="750px"
                      >
                        <Arrow__SVG
                          arrowPosition="bottom-right"
                          height="9vh"
                          showBreakpoint="750px"
                        >
                          <svg
                            className="Arrow__SVG-iWxuxU bOIamW"
                            focusable="false"
                            height="9vh"
                            preserveAspectRatio="xMinYMid meet"
                            viewBox="0 0 848 640"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M424.2 424.4L526.9 640V398.7"
                              fill="#bfe458"
                              stroke="none"
                              strokeWidth="24"
                            />
                            <path
                              d="M848 318.3L424.2 424.4V158.9"
                              fill="#aedd2c"
                              stroke="none"
                              strokeWidth="24"
                            />
                            <path
                              d="M0 0l848 318.3L636.4 0H0z"
                              fill="#bfe458"
                              stroke="none"
                              strokeWidth="24"
                            />
                          </svg>
                        </Arrow__SVG>
                      </Arrow>
                    </div>
                  </Container>
                </SpeechBubble>
              </div>
            </ContentWrapper>
          </div>
        </Wrapper>
      </PageContent>
    </AboutPage>
  </IntlProvider>
</Provider>
`;

exports[`<AboutPage /> should render and match the snapshot 1`] = `
<React.Fragment>
  <InjectIntl(HeadGear)
    messages={
      Object {
        "metaDescription": Object {
          "defaultMessage": "A short description about me and my professional life",
          "id": "portfolio.page.AboutPage.meta.description",
        },
        "pageTitle": Object {
          "defaultMessage": "ABOUT ME",
          "id": "portfolio.page.AboutPage.pageTitle",
        },
        "title": Object {
          "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
          "id": "portfolio.page.AboutPage.title",
        },
      }
    }
    path="/about"
  />
  <React.Fragment />
</React.Fragment>
`;
